cli/src/main/java/com/devonfw/tools/ide/tool/ToolCommandlet.java

```diff
@@ -481,6 +482,31 @@ protected void extract(Path file, Path targetDir) {
       } else {
         throw new IllegalStateException("Unknown archive format " + extension + ". Can not extract " + file);
       }
+
+      // collapse single parent directories
+      File currentDir = new File(targetDir.toString());
+      String[] currentSubFiles = currentDir.list();
+      if (currentSubFiles == null) {
+        throw new IllegalStateException("Extracted file does not contain any file or directories");
+      }
+      // TODO add wbin as well? (in case of msi azure installation)
+      while (currentSubFiles != null && currentSubFiles.length == 1 && !currentDir.toString().equals("bin")) {
+        currentDir = new File(String.valueOf(currentDir.toPath().resolve(currentSubFiles[0])));
```

Comment:
Stop using `File` and instead use `Path`. It needs some time to get used to it but `File` should be considered deprecated just like other JDK crap such as `Date`, `Calendar`, etc.